lib/active_record/typed_store/column.rb

```diff
@@ -30,56 +31,45 @@ def cast(value)
       casted_value
     end
 
-    def extract_default(value)
-      return value if (type == :string || type == :text) && value.nil?
-
-      type_cast(value)
-    end
-
-    def type_cast(value, map=true)
-      if array && (map || value.is_a?(Array))
-        return [] if map && !value.is_a?(Array)
-        return value.map{ |v| type_cast(v, false) }
-      end
-
-      if type == :string || type == :text
-        return value.to_s unless value.nil? && (null || array)
-      end
+    private
 
-      if IS_AR_3_2 && type == :datetime && value.is_a?(DateTime)
-        return super(value.iso8601)
-      end
+    TYPES = {
+      boolean: ::ActiveRecord::Type::Boolean,
```

Comment:
We can use `ActiveRecord::Type.lookup` in case for Rails 5 and keep own map for Rails 4.2. How does it sound for you?